src/main/java/org/testng/internal/ClassHelper.java

```diff
@@ -165,16 +166,26 @@ private static void logClassNotFoundError(String className, Exception ex) {
    * the Java access rules).
    */
   public static Set<Method> getAvailableMethods(Class<?> clazz) {
-    Set<Method> methods = Sets.newHashSet();
-    methods.addAll(Arrays.asList(clazz.getDeclaredMethods()));
+    Map<String, Set<Method>> methods = Maps.newHashMap();
+    for (final Method method : clazz.getDeclaredMethods()) {
+      if (methods.containsKey(method.getName())) {
+        methods.get(method.getName()).add(method);
+        continue;
+      }
+      methods.put(method.getName(), new HashSet<Method>() {{add(method);}});
```

Comment:
https://blog.jooq.org/2014/12/08/dont-be-clever-the-double-curly-braces-anti-pattern/